sql/sqlfmt.go

```diff
@@ -68,17 +68,22 @@ func GenerateCreateTablePrimaryKeyDefinition(pkCols []string) string {
 
 // GenerateCreateTableIndexDefinition returns index definition string for 'CREATE TABLE' statement
 // for given index. This part comes after primary key definition if there is any.
-func GenerateCreateTableIndexDefinition(isUnique, isSpatial bool, indexID string, indexCols []string, comment string) string {
+func GenerateCreateTableIndexDefinition(constraint IndexConstraint, indexID string, indexCols []string, comment string) string {
 	unique := ""
-	if isUnique {
+	if constraint == IndexConstraint_Unique {
 		unique = "UNIQUE "
 	}
 
 	spatial := ""
-	if isSpatial {
+	if constraint == IndexConstraint_Spatial {
 		unique = "SPATIAL "
 	}
-	key := fmt.Sprintf("  %s%sKEY %s (%s)", unique, spatial, QuoteIdentifier(indexID), strings.Join(indexCols, ","))
+
+	fulltext := ""
+	if constraint == IndexConstraint_Fulltext {
+		fulltext = "FULLTEXT "
+	}
+	key := fmt.Sprintf("  %s%s%sKEY %s (%s)", unique, spatial, fulltext, QuoteIdentifier(indexID), strings.Join(indexCols, ","))
```

Comment:
unique, spatial, and fulltext are all mutually exclusive?

Reply:
Good point, I'll make the change